plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/panel/GitPanelPresenter.java

```diff
@@ -0,0 +1,216 @@
+import org.eclipse.che.ide.api.event.ActivePartChangedHandler;
+import org.eclipse.che.ide.api.git.GitServiceClient;
+import org.eclipse.che.ide.api.notification.NotificationManager;
+import org.eclipse.che.ide.api.parts.PartStackType;
+import org.eclipse.che.ide.api.parts.WorkspaceAgent;
+import org.eclipse.che.ide.api.parts.base.BasePresenter;
+import org.eclipse.che.ide.api.resources.Project;
+import org.eclipse.che.ide.ext.git.client.GitEventSubscribable;
+import org.eclipse.che.ide.ext.git.client.GitEventsSubscriber;
+import org.eclipse.che.ide.ext.git.client.GitLocalizationConstant;
+import org.eclipse.che.ide.ext.git.client.GitResources;
+import org.eclipse.che.ide.ext.git.client.compare.AlteredFiles;
+import org.eclipse.che.ide.ext.git.client.compare.FileStatus.Status;
+import org.eclipse.che.ide.ext.git.client.compare.MutableAlteredFiles;
+import org.eclipse.che.ide.ext.git.client.compare.changespanel.ChangesPanelPresenter;
+import org.vectomatic.dom.svg.ui.SVGResource;
+
+/**
+ * Presenter for the left-side Git panel.
+ *
+ * @author Mykola Morhun
+ */
+@Singleton
+public class GitPanelPresenter extends BasePresenter
+    implements GitPanelView.ActionDelegate, ActivePartChangedHandler, GitEventsSubscriber {
+
+  private static final String REVISION = "HEAD";
+
+  private final GitPanelView view;
+  private final GitServiceClient service;
+  private final ChangesPanelPresenter changesPanelPresenter;
+  private final AppContext appContext;
+  private final NotificationManager notificationManager;
+  private final GitResources gitResources;
+  private final GitLocalizationConstant locale;
+
+  private Project currentProject;
+  private MutableAlteredFiles alteredFiles;
+
+  @Inject
+  public GitPanelPresenter(
+      GitPanelView view,
+      GitServiceClient service,
+      ChangesPanelPresenter changesPanelPresenter,
+      WorkspaceAgent workspaceAgent,
+      AppContext appContext,
+      EventBus eventBus,
+      GitEventSubscribable subscribeToGitEvents,
+      NotificationManager notificationManager,
+      GitResources gitResources,
+      GitLocalizationConstant locale) {
+    this.view = view;
+    this.service = service;
+    this.changesPanelPresenter = changesPanelPresenter;
+    this.appContext = appContext;
+    this.notificationManager = notificationManager;
+    this.gitResources = gitResources;
+    this.locale = locale;
+
+    this.view.setDelegate(this);
+    this.view.setChangesPanelView(this.changesPanelPresenter.getView());
+
+    this.currentProject = appContext.getRootProject();
+    this.alteredFiles = new MutableAlteredFiles(currentProject);
+
+    if (partStack == null || !partStack.containsPart(this)) {
+      workspaceAgent.openPart(this, PartStackType.NAVIGATION);
+    }
+    eventBus.addHandler(ActivePartChangedEvent.TYPE, this);
+
+    subscribeToGitEvents.addSubscriber(this);
+  }
+
+  /**
+   * Invoked each time when panel is activated.
+   */
+  private void onGitPanelOpen() {
```

Comment:
This method name means that some other panel can be opened in this presenter, may be it is better to rename the method to just `onOpen`?

Reply:
Thanks, good idea!